Cura.proto

```diff
@@ -61,6 +61,23 @@ message Polygon {
     float line_width = 3; // The width of the line being laid down
 }
 
+message LayerOptimized {
+    int32 id = 1;
+    float height = 2; // Z position
+    float thickness = 3; // height of a single layer
+
+    repeated PathSegment path_segment = 4; // layer data
```

Comment:
A very good question. I think the answer is to be safe and flexibility for future changes. The PathCompiler flushes the data when the layer changes, if the layer were to jump back to a previous layer you would get two PathSegment messages for that layer. I've not seen this happen but I'm not certain that it can't happen or won't happen in the future.